rmw_opensplice_cpp/src/rmw_publisher.cpp

```diff
@@ -170,6 +170,13 @@ rmw_create_publisher(
 
   publisher->implementation_identifier = opensplice_cpp_identifier;
   publisher->data = publisher_info;
+  publisher->topic_name = reinterpret_cast<const char *>(
+    rmw_allocate(sizeof(char) * strlen(topic_name) + 1));
```

Comment:
True.

`sizeof(char)` will evaluate to 1 on most platforms which is why it doesn't make a difference. But for completeness parentheses will be added. I will fix it here and in the place I copy-pasted it from :)